modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetCategoriesSelector.es.js

```diff
@@ -12,64 +12,77 @@
  * details.
  */
 
-import 'clay-icon';
-import Component from 'metal-component';
-import Soy from 'metal-soy';
-import {Config} from 'metal-state';
+import PropTypes from 'prop-types';
+import React from 'react';
 
-import './AssetVocabularyCategoriesSelector.es';
-import templates from './AssetCategoriesSelector.soy';
+import AssetVocabularyCategoriesSelector from './AssetVocabularyCategoriesSelector.es';
 
-/**
- * AssetCategoriesSelector is a component wrapping the existing Clay's MultiSelect component
- * that offers the user a tag selection input
- */
-class AssetCategoriesSelector extends Component {
-	_handleInputFocus(event) {
-		this.emit('inputFocus', event);
-	}
-
-	_handleSelectedItemsChange(event) {
-		this.emit('selectedItemsChange', event);
-	}
-}
+function AssetCategoriesSelector({
+	eventName,
+	groupIds,
+	id,
+	inputName,
+	onVocabulariesChange = () => {},
+	portletURL,
+	useFallbackInput,
+	vocabularies = []
+}) {
+	return (
+		<div id={id}>
+			{vocabularies.map((vocabulary, index) => {
+				const label = vocabulary.group
+					? `${vocabulary.title} ${vocabulary.group}`
+					: vocabulary.title;
 
-AssetCategoriesSelector.STATE = {
-	/**
-	 * Event name which fires when user selects a display page using item selector
-	 * @default undefined
-	 * @instance
-	 * @memberof AssetCategoriesSelector
-	 * @review
-	 * @type {?string}
-	 */
+				const isValid =
+					!vocabulary.required || vocabulary.selectedItems.length;
 
-	eventName: Config.string(),
+				return (
+					<AssetVocabularyCategoriesSelector
+						eventName={eventName}
+						groupIds={groupIds}
+						id={vocabulary.id}
```

Comment:
This is a bit misleading: `id` looks like a prop that will be assigned to a DOM element's id attribute, but what it actually is here is a string that will become *part* of an `id` attribute. One way to make this more transparent would be just to pass the actual desired id here, which (I think?) is:

```
id={`namespace_assetCategoriesSelector_${id}`}
```

Reply:
Yes that's more transparent, and it's probably a better way of "who owns the prop" ... (if that makes any sense). Thanks